tests/test_validation/test_validation_config.py

```diff
@@ -0,0 +1,520 @@
+
+import yaml
+from buildrunner.validation.config import validate_config, Errors
+
+
+def test_valid_version_config():
+    #  Invalid version
+    config = {
+        'version': 'string'
+    }
+    errors = validate_config(**config)
+    assert isinstance(errors, Errors)
+    assert errors.count() == 1
+
+    #  Valid version
+    config = {
+        'version': 2.0,
+        'steps': {
+        }
+    }
+    errors = validate_config(**config)
+    assert errors is None
+
+    # Optional version
+    config = {
+        'steps': {
+        }
+    }
+    errors = validate_config(**config)
+    assert errors is None
+
+
+def test_valid_config():
+    # Sample valid config, but not exhaustive
+    config_yaml = """
+    version: 2.0
+    steps:
+      build-container-single-platform1:
+        build:
+          path: .
+          dockerfile: Dockerfile
+          pull: false
+          platform: linux/amd64
+        push:
+          repository: mytest-reg/buildrunner-test
+          tags:
+            - latest
+      build-container-multi-platform2:
+        build:
+          path: .
+          dockerfile: Dockerfile
+          pull: false
+          platforms:
+            - linux/amd64
+            - linux/arm64
+        push:
+          repository: mytest-reg/buildrunner-test-multi-platform
+          tags:
+            - latest
+      build-container-multi-platform-push3:
+        build:
+          path: .
+          dockerfile: Dockerfile
+          pull: false
+          platforms:
+            - linux/amd64
+            - linux/arm64
+        push:
+          - myimages/image1
+          - repository: myimages/image2
+            tags:
+              - latest
+    """
+    config = yaml.load(config_yaml, Loader=yaml.Loader)
+    errors = validate_config(**config)
+    assert errors is None
+
+
+def test_multiple_errors():
+    # Multiple errors
+    # Invalid to have version as a string
+    # Invalid to have platforms and platform
+    config_yaml = """
+    version: string
+    steps:
+      build-container-multi-platform:
+        build:
+          path: .
+          dockerfile: Dockerfile
+          pull: false
+          platform: linux/amd64
+          platforms:
+            - linux/amd64
+            - linux/arm64
+        push:
+          repository: mytest-reg/buildrunner-test-multi-platform
+          tags:
+            - latest
+    """
+    config = yaml.load(config_yaml, Loader=yaml.Loader)
+    errors = validate_config(**config)
+    assert isinstance(errors, Errors)
+    assert errors.count() == 2
+
+
+def test_doc_confg():
+    # Tests the documentation example with minimal changes to make valid yaml
+    config_yaml = """
+    version: 2.0
+    steps:
+      my-build-step:
```

Comment:
Nice that you're testing the docs, but maybe this should be a separate file that we can then load into the docs as well directly?

Reply:
That is a great idea. Are you thinking that the `README.rst` would just have a link to the example file?.

Reply:
I would say it's not an exact copy and paste from the README I had to make some edits to make it valid YAML.